stage #2.5/module #3 Spring Core/#5.2. Classpath Scanning and Managed Components/content.md

```diff
@@ -0,0 +1,238 @@
+stereotype for any Spring-managed component. `@Repository`, `@Service`, and `@Controller` are specializations
+of `@Component` for more specific use cases (in the persistence, service, and presentation layers, respectively).
+Therefore, you can annotate your component classes with `@Component`, but, by annotating them with `@Repository`,
+`@Service`, or `@Controller` instead, your classes are more properly suited for processing by tools or associating with
+aspects. For example, these stereotype annotations make ideal targets for pointcuts. `@Repository`, `@Service`,
+and `@Controller` can also carry additional semantics in future releases of the Spring Framework. Thus, if you are
+choosing between using `@Component` or `@Service` for your service layer, `@Service` is clearly the better choice.
+Similarly,  `@Repository` is already supported as a marker for automatic [exception translation][1] in your persistence
+layer.
+
+## Using Meta-annotations and Composed Annotations
+
+Many of the annotations provided by Spring can be used as meta-annotations in your own code. A meta-annotation is an
+annotation that can be applied to another annotation. For example, the `@Service` annotation mentioned earlier is
+meta-annotated with `@Component`, as the following example shows:
+
+```java
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Component
+public @interface Service {
+
+    // ...
+}
+```
+
+The `@Component` causes `@Service` to be treated in the same way as `@Component`.
+
+You can also combine meta-annotations to create "composed annotations". For example, the `@RestController` annotation
+from Spring MVC is composed of `@Controller` and `@ResponseBody`.
+
+## Automatically Detecting Classes and Registering Bean Definitions
+
+Spring can automatically detect stereotyped classes and register corresponding `BeanDefinition` instances with the
+`ApplicationContext`. For example, the following two classes are eligible for such autodetection:
+
+```java
+
+@Service
+public class MusicRatingService {
+
+    private MusicRatesFinder musicRatesFinder;
+
+    public MusicRatingService(MusicRatesFinder musicRatesFinder) {
+        this.musicRatesFinder = musicRatesFinder;
+    }
+
+    // ...
+}
+
+@Repository
+public class MusicRatesFinderImpl implements MusicRatesFinder {
+    // ...
+}
+```
+
+To autodetect these classes and register the corresponding beans, you need to add `@ComponentScan` to
+your `@Configuration` class, where the `basePackages` attribute is a common parent package for the two classes.
+(Alternatively, you can specify a comma- or semicolon- or space-separated list that includes the parent package of each
+class.)
+
+```java
+
+@Configuration
+@ComponentScan(basePackages = "com.somepackage")
+public class Config {
+    // ...
+}
+```
+
+The following alternative uses XML:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
the same: concider deleting xml examples

Reply:
the same as above, some projects can still use xml config so I think it would be useful for student to show how to setup something in xml as well.